sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -29,7 +30,9 @@
  */
 class SynchronousMessageSubscriber extends BaseSubscriber<ServiceBusReceivedMessage> {
     private static final ClientLogger LOGGER = new ClientLogger(SynchronousMessageSubscriber.class);
+    private static final RuntimeException CLIENT_TERMINATED_ERROR = new RuntimeException("The receiver client is terminated. Re-create the client to continue receive attempt.");
     private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final AtomicReference<Throwable> disposalReason = new AtomicReference<>();
```

Comment:
I was wondering if we need to use atomic reference
 Writes to disposal reason are gated by isDisposed. Is volatile a better choice?

Reply:
Right, we can use volatile. I will update and also add change log